Validation/RecoTrack/src/SiStripTrackingRecHitsValid.cc

```diff
@@ -1316,19 +1249,25 @@ void SiStripTrackingRecHitsValid::createMEs(DQMStore::IBooker & ibooker,const ed
       // get detids for the layer
       // Keep in mind that when we are on the TID or TEC we deal with rings not wheel 
       int32_t lnumber = det_layer_pair.second;
+      std::string lname = det_layer_pair.first;
       std::vector<uint32_t> layerDetIds;        
-      if (det_layer_pair.first == "TIB") {
-	substructure.getTIBDetectors(activeDets,layerDetIds,lnumber,0,0,0);
-      } else if (det_layer_pair.first == "TOB") {
+      const std::string& tec = "TEC", tid = "TID", tob = "TOB", tib = "TIB";
```

Comment:
I would still argue that these could be `constexpr char[]`, or at very least their definitions should be moved outside the loop.